Makefile

```diff
@@ -21,6 +25,7 @@ format:
 
 # build and run tests
 compile:
+	make checkApi
```

Comment:
I think you don't need to call `make` but rather the function directly, maybe there's a way how to do that, if you find out how, you can remove the `make` from `buildAppleSamples` as well.
Also, when adding new functions, add it to the `.PHONY` line.